core/src/blocks/curl.rs

```diff
@@ -0,0 +1,239 @@
+            script.call("_fun", (&e,))
+        })
+        .await?
+        {
+            Ok(v) => v,
+            Err(e) => Err(anyhow!("Error in body code: {}", e))?,
+        };
+
+        let url = replace_variables_in_string(&self.url, "url", env)?;
+        let parsed_url = Url::parse(url.as_str())?;
+
+        match parsed_url.scheme() {
+            "https" => (),
+            _ => Err(anyhow!("Only the `https` scheme is authorized."))?,
+        }
+
+        let _: Vec<std::net::Ipv4Addr> = match parsed_url.host() {
+            Some(h) => match h {
+                Host::Domain(d) => {
+                    let ipv4: Vec<std::net::Ipv4Addr> = lookup_host(d)?
+                        .into_iter()
+                        .filter_map(|ip| match ip {
+                            std::net::IpAddr::V4(ip) => Some(ip),
+                            _ => None,
+                        })
+                        .collect::<Vec<_>>();
+                    match ipv4.len() {
+                        0 => Err(anyhow!("Could not find an ipv4 address for host: {}", d))?,
+                        _ => ipv4,
+                    }
+                }
+                Host::Ipv4(ip) => vec![ip],
+                Host::Ipv6(_) => Err(anyhow!("Ipv6 addresses are not supported."))?,
+            },
+            None => Err(anyhow!("Provided URL has an empty host"))?,
+        }
+        .into_iter()
+        .map(|ip| {
+            lazy_static! {
+                static ref RE: Regex = Regex::new(r"^(0|127|10|172|192)\..*").unwrap();
+            }
+            match RE.is_match(ip.to_string().as_str()) {
+                true => Err(anyhow!("Forbidden IP range"))?,
+                false => Ok(ip),
+            }
+        })
+        .collect::<Result<Vec<_>>>()?;
+
+        // TODO(spolu): encode query
+
+        let mut req = Request::builder().method(method).uri(url.as_str());
+
+        let headers = req.headers_mut().unwrap();
+        match headers_value {
+            Value::Object(h) => {
+                for (key, value) in h {
+                    match value {
+                        Value::String(value) => {
+                            headers.insert(
+                                header::HeaderName::from_bytes(key.as_bytes())?,
+                                header::HeaderValue::from_bytes(value.as_bytes())?,
+                            );
+                        }
+                        _ => Err(anyhow!("Header value for header {} must be a string", key))?,
+                    }
+                }
+            }
+            _ => Err(anyhow!(
+                "Returned headers must be an object with string values.",
+            ))?,
+        };
+
+        let req = match body_value {
+            Value::String(body) => req.body(Body::from(body))?,
+            Value::Null => req.body(Body::empty())?,
+            _ => Err(anyhow!("Returned body must be either a string or null."))?,
```

Comment:
This means that devs have to use `JSON.stringify` in their body function if they are sending JSON, yes? If so, we might want to put a warning about that in the UI. Or alternately, we could do it automatically when they set the content type?

Reply:
Agreed we could do something smarter, but will defer to later to iterate on that.